src/compiler/transformers/ts.ts

```diff
@@ -3358,5 +3355,11 @@ namespace ts {
 
             return isPropertyAccessExpression(node) || isElementAccessExpression(node) ? resolver.getConstantValue(node) : undefined;
         }
+
+        function shouldEmitAliasDeclaration(node: Node): boolean {

```

Comment:
Why extract `importsNotUsedAsValues` at the top of the transformer, but not `preserveValueImports`?

Reply:
(sorry, this comment should have been associated with line following this one)